### PR TITLE
Role Welcome UI

### PR DESCRIPTION
**Pull request by Ephememory**
https://cdn.discordapp.com/attachments/1057060299947114498/1063169693399130172/text.png

### CODE REVIEW

**Comment by matekdev** [comment]
we chatted about this i'll have a PR up with some of these changes

**Comment by matekdev** [comment]
Solves https://github.com/CigarLounge/sbox-TTT/issues/178

File: code/Events/Event.Roles.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+		/// <summary>
+		/// Fired when all roles have been assigned.
+		/// </summary>
+		public class AssignedAttribute : EventAttribute
+		{
+			public AssignedAttribute() : base( Assigned ) { }
+		}
+	}
+}
```
Comment (kolexxx) [inline code/Events/Event.Roles.cs:19]: Why is this needed? We should be using `GameEvent.Round.Start`.

File: code/Util/NetworkedEvent.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+using Sandbox;
+
+namespace TTT;
+
+public static partial class NetworkedEvent
```
Comment (kolexxx) [inline code/Util/NetworkedEvent.cs:5]: This is not needed.

**Review by kolexxx — CHANGES_REQUESTED** [review]

**Comment by matekdev** [comment]
I'll branch off of this and finish up the rest, thanks.

**Pull request closed without merging**